opendevin/controller/agent_controller.py

```diff
@@ -306,7 +307,8 @@ async def _step(self):
             await self.report_error(str(e))
             return
 
-        logger.info(action, extra={'msg_type': 'ACTION'})
+        if not isinstance(action, MessageAction):
```

Comment:
Why only log non-message action ?

Reply:
https://github.com/OpenDevin/OpenDevin/pull/2035#issuecomment-2179416291

Reply:
I think there is a duplicate MessageAction indeed, we log both USER and AGENT source in `on_event`, and here too. `on_event` is always hit when any MessageAction is added to the event stream

Reply:
ahhh got it! Thanks for the explanation!